### PR TITLE
gcc 14.1.0 gcc@13 13.2.0 (new formula)

### PR DESCRIPTION
**Pull request by fxcoudert**
GCC 14.1 release notes: https://gcc.gnu.org/pipermail/gcc/2024-April/243853.html

This does not include the ARM-specific changes, so it is expected to fail. But it will be useful to see what it reports on Intel.

### CODE REVIEW

**Comment by fxcoudert** [comment]
Companion brew PR: https://github.com/Homebrew/brew/pull/17202

**Comment by fxcoudert** [comment]
Second GCC 14.1 Release Candidate
https://gcc.gnu.org/pipermail/gcc/2024-May/243901.html

**Review by carlocab — APPROVED** [review]
Still draft?

File: Formula/g/gcc.rb [hunk 1]
```diff
@@ -5,27 +5,16 @@ class Gcc < Formula
   head "https://gcc.gnu.org/git/gcc.git", branch: "master"
 
   stable do
-    url "https://ftp.gnu.org/gnu/gcc/gcc-13.2.0/gcc-13.2.0.tar.xz"
-    mirror "https://ftpmirror.gnu.org/gcc/gcc-13.2.0/gcc-13.2.0.tar.xz"
-    sha256 "e275e76442a6067341a27f04c5c6b83d8613144004c0413528863dc6b5c743da"
+    url "https://gcc.gnu.org/pub/gcc/snapshots/14.1.0-RC-20240503/gcc-14.1.0-RC-20240503.tar.xz"
+    version "14.1.0"
```
Comment (carlocab) [inline Formula/g/gcc.rb:9]: Oh, right. Still RC.

Comment (egorpugin) [reply Formula/g/gcc.rb:9]: There are release snapshots on the mirrors already
http://ftp.ntua.gr/mirror/gnu/gcc/releases/gcc-14.1.0/

**Comment by fxcoudert** [comment]
GCC 14.1 is released: https://gcc.gnu.org/pipermail/gcc/2024-May/243921.html

**Review by chenrui333 — APPROVED** [review]

**Review by carlocab — APPROVED** [review]

**Review by carlocab — APPROVED** [review]

**Comment by fxcoudert** [comment]
I have no idea why the patch checksums would work on Linux and Apple ARM, but fail on Apple Intel.

**Comment by fxcoudert** [comment]
Checksum has apparently changed over time. So I'm switching from `patch` to `commitdiff_plain` URLs.

**Review by carlocab — APPROVED** [review]

**Comment by fxcoudert** [comment]
- test failure of `abi-compliance-checker`: will investigate separately
- test failure of `mrbayes`: `OpenCL error: CL_INVALID_VALUE` on macOS 12 ARM only. I think it's a set up issue of the VM there.

**Pull request merged**